src/services/accounts/AccountsAssetsService.ts

```diff
@@ -125,12 +147,47 @@ export class AccountsAssetsService extends AbstractService {
 					address
 				);
 
+				/**
+				 * The following checks for three different cases:
+				 *
+				 * 1. Via runtime v9160 the updated storage introduces a `reason` field,
+				 * and polkadot-js wraps the newly returned `PalletAssetsAssetAccount` in an `Option`.
+				 *
+				 * 2. `query.assets.account()` return `PalletAssetsAssetBalance` which exludes `reasons` but has
+				 * `sufficient` as a key.
+				 *
+				 * 3. The older legacy type of `PalletAssetsAssetBalance` has a key of `isSufficient` instead
+				 * of `sufficient`.
+				 *
+				 */
+				let balance = null,
+					isFrozen = null,
+					isSufficient = null;
+				if (assetBalance.isSome) {
```

Comment:
This section feels unreadable to me just based off the linter alone. I propose we increase the characters per line for some more readability in a general sense.

Reply:
I found this code a bit harder to read, and was sortof wondering whether duplicating a little more might make it easier to grok; like moving any variables needed into each branch and returning the resolt from each?

I had a quick go at it; something like this perhaps:

```
const assetBalance = await historicApi.query.assets.account(
	assetId,
	address
);

/**
 * The following checks for three different cases:
 */

	// 1. Via runtime v9160 the updated storage introduces a `reason` field,
	// and polkadot-js wraps the newly returned `PalletAssetsAssetAccount` in an `Option`.
if (assetBalance.isSome) {
	const balanceProps = assetBalance.unwrap();

	return {
		assetId,
		balance: balanceProps.balance,
		isFrozen: balanceProps.isFrozen,
		isSufficient: balanceProps.reason.isSufficient
	}
}

// 2. `query.assets.account()` return `PalletAssetsAssetBalance` which exludes `reasons` but has
// `sufficient` as a key.
if ((assetBalance as unknown as PalletAssetsAssetBalance).sufficient) {
	const balanceProps = assetBalance as unknown as PalletAssetsAssetBalance;

	return {
		assetId,
		balance: balanceProps.balance,
		isFrozen: balanceProps.isFrozen,
		isSufficient: balanceProps.sufficient
	}
}

// 3. The older legacy type of `PalletAssetsAssetBalance` has a key of `isSufficient` instead
// of `sufficient`.
if (assetBalance['isSufficient'] as bool) {
	const balanceProps = assetBalance as unknown as LegacyPalletAssetsAssetBalance;

	return {
		assetId,
		balance: balanceProps.balance,
		isFrozen: balanceProps.isFrozen,
		isSufficient: balanceProps.isSufficient
	}
}

// Fallback; should we throw if we get here?
return {
	assetId,
	balance: null,
	isFrozen: null,
	isSufficient: null,
};
```

Not a big deal either way; mainly I was interested to see if it could be cleaned up at all :)

Reply:
Updated to your snippet above. Thanks again, greatly improved readability.

Reply:
So for the last part, I was on the same though train where throwing an error would make sense, but just remembered that polkadot-js always returns the values back as 0<u128>, and false<bool> if they don't exist. So I changes it slightly to reflect that